### ISSUE TITLE
ppxlib#trunk-support on 5.1.0~alpha1 breaks `let f: type a.` syntax

### ISSUE THREAD

**Issue by copy** [ocaml-ppx/ppxlib#418]
Consider:

```
let f: type a. a option -> _ = assert false
```

```
(executable
 (name test)
 (flags :standard -dsource)
 (preprocess (pps ppx_here)))
```

On OCaml 5.1.0~alpha1 it results in the following error:

```
File "dune", line 1, characters 0-82:  
1 | (executable
2 |  (name test)
3 |  (flags :standard -dsource)
4 |  (preprocess (pps ppx_here)))
[@@@ocaml.ppx.context
  {
    tool_name = "ppx_driver";
    include_dirs = [];
    load_path = [];
    open_modules = [];
    for_package = None;
    debug = false;
    use_threads = false;
    use_vmthreads = false;
    recursive_types = false;
    principal = false;
    transparent_modules = false;
    unboxed_types = false;
    unsafe_string = false;
    cookies = []
  }]
let (f : 'a . a option -> _) = fun (type a) -> (assert false : a option -> _)
File "test.ml", line 1, characters 15-16:
1 | let f: type a. a option -> _ = assert false
                   ^
Error: Unbound type constructor a
```

While on 5.0:

```
let f : type a. a option -> _ = assert false
```

(any ppx that uses ppxlib demonstrates this issue, ppx_here is just a small example)

**Comment by copy** [comment]
Here's a simpler one:

```
File "test.ml", line 1, characters 4-10:
1 | let (x, y) : (int * int) = assert false
        ^^^^^^
Error: This pattern matches values of type 'a * 'b
       but a pattern was expected which matches values of type int * int
```

**Comment by panglesd** [comment]
Thanks for the report. This is related to the parsetree migration (see https://github.com/ocaml-ppx/ppxlib/pull/407). I'll have a look.

**Comment by panglesd** [comment]
Should be fixed by #420.

**Comment by pitag-ha** [comment]
Thanks again for reporting, @copy! It's fixed now.
